pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -246,6 +249,14 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 				// If it fails it means the test from the previous patch failed
 				// so we can safely replace this node in the CNP status.
 				replaceCEPStatus := []k8s.JSONPatch{
+					// If the stored UID matches the one in the ciliumendpoint then
+					// this first patch is a no-op, otherwise the entire patch will
+					// not be applied as uid is immutable.
+					{
+						OP:    "test",
+						Path:  "/metadata/uid",
+						Value: e.GetCiliumEndpointUID(),
```

Comment:
@tommyp1ckles out of curiosity, what happens if we are starting a cilium agent for which we don't know the CEP UID? For example a Cilium agent restores endpoints that have an empty CEP UID set.

Reply:
This situation is captured here: https://github.com/cilium/cilium/pull/21768/files#diff-6c96962323677414649840871089f13cad859f12b2b42ca62bcf317b49f69571R319

The idea is, that we can safely "take ownership" of a CEP when restoring from an empty UID reference only if that CEP was created locally.